tests/components/switchbot_cloud/test_cover.py

```diff
@@ -0,0 +1,487 @@
+"""Test for the switchbot_cloud Cover."""
+
+from unittest.mock import patch
+
+from switchbot_api import Device
+
+from homeassistant.components.cover import DOMAIN as COVER_DOMAIN
+from homeassistant.components.switchbot_cloud import SwitchBotAPI
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.const import (
+    ATTR_ENTITY_ID,
+    SERVICE_CLOSE_COVER,
+    SERVICE_CLOSE_COVER_TILT,
+    SERVICE_OPEN_COVER,
+    SERVICE_OPEN_COVER_TILT,
+    SERVICE_SET_COVER_POSITION,
+    SERVICE_SET_COVER_TILT_POSITION,
+    SERVICE_STOP_COVER,
+)
+from homeassistant.core import HomeAssistant
+
+from . import configure_integration
+
+
+async def test_set_attributes_coordinator_is_none(
+    hass: HomeAssistant, mock_list_devices, mock_get_status
+) -> None:
+    """test_set_attributes_coordinator_is_none."""
+    mock_list_devices.return_value = [
+        Device(
+            version="V1.0",
+            deviceId="cover-id-1",
+            deviceName="cover-1",
+            deviceType="Curtain",
+            hubDeviceId="test-hub-id",
+        ),
+    ]
+
+    mock_get_status.side_effect = [None, None]
+
+    entry = await configure_integration(hass)
+
+    assert entry.state is ConfigEntryState.LOADED
+
+
+async def test_blindtilt_set_attributes_coordinator_is_none(
+    hass: HomeAssistant, mock_list_devices, mock_get_status
+) -> None:
+    """test_set_attributes_coordinator_is_none."""
+    mock_list_devices.return_value = [
+        Device(
+            version="V1.0",
+            deviceId="cover-id-1",
+            deviceName="cover-1",
+            deviceType="Blind Tilt",
+            hubDeviceId="test-hub-id",
+        ),
+    ]
+
+    mock_get_status.side_effect = [None, None]
+
+    entry = await configure_integration(hass)
+
+    assert entry.state is ConfigEntryState.LOADED
+
+
+async def test_blindtilt_set_attributes_position(
+    hass: HomeAssistant, mock_list_devices, mock_get_status
+) -> None:
+    """test_set_attributes_coordinator_is_none."""
+    mock_list_devices.return_value = [
+        Device(
+            version="V1.0",
+            deviceId="cover-id-1",
+            deviceName="cover-1",
+            deviceType="Blind Tilt",
+            hubDeviceId="test-hub-id",
+        ),
+    ]
+
+    mock_get_status.side_effect = [
+        {
+            "slidePosition": 95,
+        },
+        {
+            "slidePosition": 90,
+        },
+    ]
+
+    entry = await configure_integration(hass)
+
+    assert entry.state is ConfigEntryState.LOADED
```

Comment:
I am not sure what we test here, we just verify that the integration is able to set up, but I feel like you are testing something more with regards to the state of the entities it creates

Reply:
I have rewrite test code ,please take a look